python/chronos/src/bigdl/chronos/data/utils/prometheus_df.py

```diff
@@ -0,0 +1,193 @@
+
+    def _do_query(self, path: str, params: dict) -> dict:
+        resp = self.http.get(urljoin(self.api_url, path), params=params)
+        if resp.status_code not in [400, 422, 503]:
+            resp.raise_for_status()
+
+        response = resp.json()
+        if response['status'] != 'success':
+            invalidInputError(False, "Fail to collect data from Prometheus! "
+                              "{errorType}: {error}".format_map(response))
+
+        return response['data']
+
+
+def to_pandas(data: dict) -> Union[Matrix, Vector, Scalar, String]:
+    """Convert Prometheus data object to Pandas object."""
+    result_type = data['resultType']
+    if result_type == 'vector':
+        return pd.Series((np.float64(r['value'][1]) for r in data['result']),
+                         index=(metric_name(r['metric']) for r in data['result']))
+    elif result_type == 'matrix':
+        return pd.DataFrame({
+            metric_name(r['metric']):
+                pd.Series((np.float64(v[1]) for v in r['values']),
+                          index=(pd.Timestamp(v[0], unit='s') for v in r['values']))
+            for r in data['result']})
+    elif result_type == 'scalar':
+        return np.float64(data['result'])
+    elif result_type == 'string':
+        return data['result']
+    else:
+        invalidInputError(False, f"The collected Prometheus data is unknown type {result_type}.")
+
+
+def metric_name(metric: dict) -> str:
+    """Convert metric labels to standard form."""
+    name = metric.get('__name__', '')
+    labels = ','.join(('{}={}'.format(k, json.dumps(v)) for k, v in metric.items()
+                      if k != '__name__'))
+    return '{0}{{{1}}}'.format(name, labels)
+
+
+def _timestamp(value):
+    if isinstance(value, datetime.datetime):
+        return value.timestamp()
+    else:
+        return value
+
+
+def _duration(value):
+    if isinstance(value, datetime.timedelta):
+        return value.total_seconds()
+    else:
+        return value
+
+
+def GetRangeDataframe(prometheus_url, query, starttime, endtime, step, columns, **kwargs):
+    '''
+    Convert the Prometheus data over the specified time period to dataframe and confirm
+    dt_col, target_col, id_col and extra_feature_col.
+
+    Return dataframe and col names.
+    '''
+    from bigdl.chronos.data.utils.utils import _check_type
+    _check_type(prometheus_url, "prometheus_url", str)
+    _check_type(query, "query", str)
+    _check_type(starttime, "starttime", (str, float))
+    _check_type(endtime, "endtime", (str, float))
+    _check_type(step, "step", (str, float))
+
+    # Generate dateframe
+    Pro_client = Prometheus(prometheus_url)
+    Pro_df = Pro_client.query_range(query, starttime, endtime, step, **kwargs)
+    df = pd.DataFrame(columns=Pro_df.columns.tolist())
+    df.insert(0, "datetime", Pro_df.index)
+    for col in Pro_df.columns.tolist():
+        df[col] = Pro_df[col].values
```

Comment:
preferably, we don't use capital latter for variables (but we don't need to change it now)